src/core/etl/src/Flow/ETL/PHP/Type/TypeFactory.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\PHP\Type;
+
+use Flow\ETL\Exception\InvalidArgumentException;
+use Flow\ETL\PHP\Type\Logical\List\ListElement;
+use Flow\ETL\PHP\Type\Logical\ListType;
+use Flow\ETL\PHP\Type\Logical\Map\MapKey;
+use Flow\ETL\PHP\Type\Logical\Map\MapValue;
+use Flow\ETL\PHP\Type\Logical\MapType;
+use Flow\ETL\PHP\Type\Logical\Structure\StructureElement;
+use Flow\ETL\PHP\Type\Logical\StructureType;
+use Flow\ETL\PHP\Type\Native\ArrayType;
+use Flow\ETL\PHP\Type\Native\NullType;
+use Flow\ETL\PHP\Type\Native\ObjectType;
+use Flow\ETL\PHP\Type\Native\ScalarType;
+
+final class TypeFactory
+{
+    public function getType(mixed $value) : Type
+    {
```

Comment:
I think you should check if it's not a resource or callable as well

Reply:
You mean by explicit? If all checks fail, it will end-up with an exception at the end, so IMO it's covered already (ofc can add a test for it).

Reply:
nah, I mean that we should have a type for Resource and Callables since array of resources or array of callables or list/map of resource/callables is something that should be supported.